price-feeder/oracle/provider/okx.go

```diff
@@ -157,15 +186,28 @@ func (p *OkxProvider) messageReceived(messageType int, bz []byte) {
 		return
 	}
 
-	var tickerResp OkxTickerResponse
+	var (
+		tickerResp OkxTickerResponse
+		candleResp OkxCandleResponse
+	)
+
+	// sometimes the message received is not a ticker or a candle response.
 	if err := json.Unmarshal(bz, &tickerResp); err != nil {
-		// sometimes it returns other messages which are not tickerResponses
-		p.logger.Err(err).Msg("could not unmarshal")
-		return
+		p.logger.Debug().Err(err).Msg("could not unmarshal ticker resp")
+	}
+	if err := json.Unmarshal(bz, &candleResp); err != nil {
+		p.logger.Debug().Err(err).Msg("could not unmarshal candle resp")
 	}
 
-	for _, tickerPair := range tickerResp.Data {
-		p.setTickerPair(tickerPair)
+	if tickerResp.ID.Channel == "tickers" {
+		for _, tickerPair := range tickerResp.Data {
+			p.setTickerPair(tickerPair)
+		}
+	}
+	if candleResp.ID.Channel == "candle1m" {
+		for _, candlePair := range candleResp.Data {
+			p.setCandlePair(candlePair, candleResp.ID.InstID)
+		}
```

Comment:
This bit of logic is necessary for okx since if the response is a ticker response, it will unmarshal into the `candleResp.Data` as a string array where [0] is an empty string array.

We could handle it by checking lengths, but this seemed more straightforward